src/page/Variant.tsx

```diff
@@ -85,17 +90,68 @@ class Variant extends React.Component<IVariantProps> {
         );
     }
 
-    private getMutationMapper() {
+    @computed
+    private get getMutationMapperStore() {
         let mutation = this.variantToMutation(this.annotationSummary);
         if (
             mutation[0] &&
             mutation[0].gene &&
             mutation[0].gene.hugoGeneSymbol.length !== 0
+        ) {
+            return initDefaultMutationMapperStore({
+                genomeNexusUrl: genomeNexusApiRoot,
+                data: mutation,
+                hugoSymbol: getTranscriptConsequenceSummary(
+                    this.annotationSummary
+                ).hugoGeneSymbol,
+                oncoKbUrl: 'https://www.cbioportal.org/proxy/oncokb',
+                // select the lollipop by default
+                selectionFilters: [
+                    {
+                        type: DataFilterType.POSITION,
+                        values: [mutation[0].proteinPosStart],
+                    },
+                ],
+            });
+        }
+        return undefined;
+    }
+
+    @computed get allValidTranscripts() {
+        if (
+            this.getMutationMapperStore!.transcriptsWithAnnotations.result &&
+            this.getMutationMapperStore!.transcriptsWithAnnotations.result
+                .length > 0
+        ) {
+            return this.getMutationMapperStore!.transcriptsWithAnnotations
+                .result;
+        }
+        return [];
+    }
+
+    @action.bound
+    private setActiveTranscript(transcriptId: string) {
```

Comment:
would be nice if the active transcript is a URL parameter as well. So clicking the transcript changes the url parameter which in turn changes the visualization. I'm not sure what would be a good way to implement, but Adam, Onur or Aaron might have some suggestion. There should be some examples in cbioportal-frontend